File: src/main/java/br/com/useinet/finance/dto/ContaResponse.java
package br.com.useinet.finance.dto;

import br.com.useinet.finance.model.Conta;

public class ContaResponse {
    private Long id;
    private String nome;
    private Double saldo;

    public static ContaResponse from(Conta c) {
        ContaResponse r = new ContaResponse();
        r.id = c.getId();
        r.nome = c.getNome();
        r.saldo = c.getSaldo();
        return r;
    }

    public Long getId() { return id; }
    public String getNome() { return nome; }
    public Double getSaldo() { return saldo; }
}

File: src/main/kotlin/br/com/useinet/finance/dto/ContaResponse.kt
package br.com.useinet.finance.dto

import br.com.useinet.finance.model.Conta

data class ContaResponse(val id: Long, val nome: String, val saldo: Double) {
    companion object {
        fun from(c: Conta) = ContaResponse(
            id = requireNotNull(c.id) { "Conta.id não pode ser nulo" },
            nome = requireNotNull(c.nome) { "Conta.nome não pode ser nulo" },
            saldo = requireNotNull(c.saldo) { "Conta.saldo não pode ser nulo" }
        )
    }
}
